project.py

```diff
@@ -147,19 +168,31 @@ def run_tasks(tasks: Sequence[Task]) -> List[TaskResult]:
   return results
 
 
-def build_tasks(args: argparse.Namespace) -> List[Task]:
-  '''Build a list of tasks based on the parsed arguments.'''
+def create_tasks(args: argparse.Namespace) -> List[Task]:
+  '''Create a list of tasks based on the parsed arguments.'''
   tasks = []
   if args.setup:
-    tasks.append(Task('Set up and ensure dependencies', setup_environment))
+    plan = SetupPlan(
+      install_dependencies=True,
+      required_tools=[Tool('cmake'), Tool('make')],
+      cpp_compilers=['clang++', 'g++', 'clang++18', 'g++14'],
+      cpp_standards=['c++2b'],
```

Comment:
I somewhat believe that we should make the standard a CONSTANT, and use `./project.py --cpp-std` to query the C++ standard. CMake can read the standard from env var. Not sure if this is doable though.